apps/authentication-service/cypress/e2e/express.cy.ts

```diff
@@ -18,4 +18,23 @@ describe("Express", () => {
       .its("status")
       .should("eq", 404);
   });
+
+  it("tests rate limiting middleware", () => {
+    for (let i = 0; i < 8; i++) {
+      if (i < 3) {
+        cy.request({
+          url: `http://localhost:${PORT}/ping`,
+        })
+          .its("status")
+          .should("eq", 200);
+      } else {
+        cy.request({
+          url: `http://localhost:${PORT}/ping`,
+          failOnStatusCode: false,
+        })
+          .its("status")
+          .should("eq", 429);
+      }
+    }
+  });
```

Comment:
I'm not sure if this test is useful. It does seem to test the rate-limiter live, but it might be brittle. It relies on very specific numbers in order to work, which I guess isn't that big of a deal.

I'd love feedback on a better way to test this. I couldn't find anything useful online.

Reply:
From what I can gather from the test the rate limit gets hit after 3 api requests in some small amount of time. I don't feel like the exact numbers are very important to test or have defined in a test. You could make the test description a little better like: "rate limit is reached after many quick requests in under <some amount of time>"

Then probably a safer test would probably be something along the lines of 
```javascript
expect {
  for (let i = 0; i < 10; i++) {
    cy.request({url: ...})
}.to throw an error with 429 status
```

I think you could do something like that, if you don't use `failOnStatusCode` maybe it will throw the error.